### PR TITLE
Fixes questionnaire filtering on evaluation creation

### LINKED ISSUE #1411: Questionnaire filtering on Evaluation creation broken
When creating a new Evaluation in the staff create evaluation form, in the general questionnaire list questionnaires which are set to be hidden but have never been used are listed. They should be hidden instead.

### PR DESCRIPTION
**Pull request by niklasmohrin**
Fixes #1411 

I believe this is a fix, but I could use a confirmation by someone.

### CODE REVIEW

**Comment by niklasmohrin** [comment]
Conclusion:
- `Questionnaire.objects.filter(contribution__evaluation=None).all()` is not empty, but contains the questionnaire "Master Thesis", a questionnaire with 0 uses
- in `EvaluationForm.__init__`, `self.instance` is not `None`, but a `Evaluation` with no `id` or `pk`
- somehow, the "Master thesis" questionnaire matches against the unsaved `Evaluation` (`self.instance`)

**Comment by He3lixxx** [comment]
Hmm, best way to verify that you fixed something is simply writing a regression test. Then, there is also no doubt that you actually fixed the problem.

File: evap/staff/tests/test_forms.py [hunk 1]
```diff
@@ -681,3 +681,30 @@ def test_locked_questionnaire(self):
...
+            form_data["general_questionnaires"]
+        )
+
+        form_data["general_questionnaires"] = [questionnaire2.pk]
+        form = EvaluationForm(form_data, instance=evaluation, semester=semester)
+        form.save()
+
+        form_data = get_form_data_from_instance(EvaluationForm, evaluation)
+
+        # Now, that no evaluation is using the first questionnaire, it should not be visible anymore
```
Comment (janno42) [inline evap/staff/tests/test_forms.py:706]: well. the questionnaire needs to be set to hidden if you want to test the case described in the issue :)
in form_data you will find only selected questionnaires, so you should check for the options or just the existence of the questionnaire's name on the webpage.

File: evap/staff/tests/test_forms.py [hunk 2]
```diff
@@ -681,3 +681,29 @@ def test_locked_questionnaire(self):
...
+
+        # Delete the hidden questionnaire
+        form.save()
+
+        # Now, that no evaluation is using the first questionnaire, it should not be visible anymore
+        form = EvaluationForm(instance=evaluation, semester=evaluation.course.semester)
+        self.assertNotIn(
+            questionnaire,
+            form.fields["general_questionnaires"].queryset
+        )
```
Comment (He3lixxx) [inline evap/staff/tests/test_forms.py:709]:
```python
form = EvaluationForm(instance=evaluation, semester=evaluation.course.semester)
self.assertNotIn(questionnaire, form.fields["general_questionnaires"].queryset)

evaluation.general_contribution.questionnaires.add(questionnaire)
form = EvaluationForm(instance=evaluation, semester=evaluation.course.semester)
self.assertIn(questionnaire, form.fields["general_questionnaires"].queryset)
```

**Comment by He3lixxx** [comment]
Check out codacy, it says you have a blank line at the end of a file

**Pull request merged**